File: accelerators/aks-sb-azmonitor-microservices/src/cargo-processing-api/src/main/java/com/microsoft/cse/cargoprocessing/api/chaos/impl/ServiceBusKillingMonkey.java
package com.microsoft.cse.cargoprocessing.api.chaos.impl;

import java.util.Map;

import org.springframework.stereotype.Service;

import com.azure.messaging.servicebus.ServiceBusSenderClient;

@Service
public class ServiceBusKillingMonkey extends BaseMonkey {
    public ServiceBusKillingMonkey() {
        super("service-bus-failure");
    }

    @Override
    public void WakeTheMonkey(Map<String, Object> parameters) {
        // Oh, let's just close that sender before trying to use it, what could possibly
        // go wrong?
        ServiceBusSenderClient sender = getParm(parameters, "sender", null);
        sender.close();
    }
}

File: accelerators/aks-sb-azmonitor-microservices/src/cargo-processing-validator/src/chaos/ServiceBusKillingMonkey.ts
import { ServiceBusSender } from '@azure/service-bus';
import { ServiceBusReceiverWithTelemetry } from '../services/ServiceBusWithTelemetry';
import { ChaosMonkey } from "./ChaosMonkey";

export class ServiceBusKillingMonkey extends ChaosMonkey {
    constructor(queueReceiver: ServiceBusReceiverWithTelemetry) {
        super("service-bus-failure", queueReceiver);
    }

    async wakeTheMonkey(parameters: Map<string, object>): Promise<void> {
        const sender = parameters.get("sender") as ServiceBusSender;
        await sender.close();
    }
}